c++/src/Statistics.hh

```diff
@@ -106,15 +106,50 @@ namespace orc {
 
     void setMinimum(T min) { _minimum = min; }
 
-    // GET / SET valueCount_
+    // GET / SET _valueCount
     uint64_t getNumberOfValues() const { return _valueCount; }
 
     void setNumberOfValues(uint64_t numValues) { _valueCount = numValues; }
 
-    // GET / SET hasNullValue_
+    // GET / SET _hasNullValue
     bool hasNull() const { return _hasNull; }
 
     void setHasNull(bool hasNull) { _hasNull = hasNull; }
+
+    void reset() {
+      _hasNull = false;
+      _hasMinimum = false;
+      _hasMaximum = false;
+      _hasSum = false;
+      _hasTotalLength = false;
+      _totalLength = 0;
+      _valueCount = 0;
+    }
+
+    // sum is not merged here as we need to check overflow
+    void merge(const InternalStatisticsImpl& other) {
+      _hasNull = _hasNull || other._hasNull;
+      _valueCount += other._valueCount;
+
+      if (other._hasMinimum) {
+        if (!_hasMinimum) {
+          _hasMinimum = _hasMaximum = true;
+          _minimum = other._minimum;
+          _maximum = other._maximum;
+        } else {
+          // all template types should support operator<
+          if (_maximum < other._maximum) {
```

Comment:
I've fixed it and created a new PR.  I have removed "static" identifier as this template function is used in several modules including UTs. Please review it again. Thanks!